### ISSUE TITLE
Missing item in Index

### ISSUE THREAD

**Issue by Intelligent2013** [metanorma/metanorma#298]
Source issue: https://github.com/metanorma/stepmod2mn/issues/36

The source document: 
[activity.zip](https://github.com/metanorma/metanorma/files/11612312/activity.zip)

In the adoc there are one index entry `activity` and three `Activity`:

`03-terms.adoc`:
```
===== [[term-activity]]activity (((activity,term)))
```
`04-info_reqs.adoc`:
```
===== Activity (((Activity,ARM object definition)))
```
`05-mim.adoc`:
```
==== <<activity_arm.activity,Activity>> (((Activity,mapping specification)))
```
`93-arm-express-g.adoc`:
```
.ARM entity level EXPRESS-G diagram 1 of 1 (((Activity_method,ARM EXPRESS-G)))(((activity_item,ARM EXPRESS-G)))(((Activity,ARM EXPRESS-G)))(((Activity_relationship,ARM EXPRESS-G)))(((Activity_status,ARM EXPRESS-G)))(((Applied_activity_assignment,ARM EXPRESS-G)))
```

The command: `bundle exec metanorma -t iso -x presentation,html document.adoc`

`Index` contains only `activity...term`:
![image](https://github.com/metanorma/metanorma/assets/5559926/68847462-db2b-4614-98ab-e963410c5790)

But when I remove `(((activity,term)))` from `03-terms.adoc`, then `Index` contains `Activity ...`:
![image](https://github.com/metanorma/metanorma/assets/5559926/a40f35b2-f55e-4af4-9c7d-4fcb23362309)

Expected result - original HTML:
![image](https://github.com/metanorma/metanorma/assets/5559926/7b6b583b-8b1b-4716-b746-535443e32ce5)

**Comment by stuartgalt** [comment]
@ronaldtse This still seems to be a bug.  Should it be moved to a different repo?  Also **Veronique considers this a blocker** for being able to publish.  The latest SMRL (old way) contains both the link to the object in the schema as well as where it shows up in an EXPRESS-G diagram.  The latest SUMA builds just include a link to where the the object is described in the text.

**Comment by opoudjis** [comment]
Diagnosed. This is a defect in the index builder in isodoc, not in term
processing: to sort the index case-insensitively, entries are keyed on
their downcased form, and two entries differing only in case collide on
that key — one of them is silently discarded **together with all its
subentries**. That is why `activity → term` survives while all three
`Activity` subentries (ARM object definition, mapping specification,
ARM EXPRESS-G) vanish, and why removing `(((activity,term)))` makes
`Activity` reappear. The current behaviour is neither folding nor
differentiation; it is data loss, and it will be fixed. (@stuartgalt:
the fix lands in metanorma/isodoc, with a mirror in metanorma-bipm
which overrides the same code; this issue can stay here as the tracking
ticket.)

However, the expected output shown here — `Activity` and `activity` as
*separate* index headwords — is not going to become the default. For
general prose, case folding is the correct behaviour: `((activity))` in
mid-sentence and `((Activity))` at the start of a sentence are the same
word and must merge into one entry. Prose authors would be
surprised by an index that splits headwords on capitalisation, and they
are the ones best served by a default that just works. Case-significant
headwords belong to identifier-heavy documents like ISO 10303, where
`Activity` is a case-bearing EXPRESS entity name rather than a
capitalised word — and the STEP community, which knows exactly why its
identifiers are case-significant, is well placed to switch that
behaviour on deliberately.

Proposal:

1. **Default: genuine case folding.** Entries differing only in case
   merge into a single headword — lowercase spelling wins — with the
   union of their subentries and locators. For this document that
   gives:

   ```
   activity
       term, 3.2.1
       ARM object definition, 4.3.1
       ARM EXPRESS-G, C.2
       mapping specification, 5.1.1
   ```

   Nothing is lost; nothing splits on case.

2. **Opt-in: case-sensitive indexing** via a document attribute as a
   processing option (e.g. `:index-case-sensitive:`). With it set,
   case-distinct headwords are kept apart and sorted adjacently,
   reproducing the SMRL-style index in the expected screenshot.
   stepmod2mn should set this attribute in the documents it generates,
   which restores the legacy behaviour for ISO 10303 without imposing
   it on anyone else.

🤖

**Comment by opoudjis** [comment]
Implemented as proposed:

- metanorma/isodoc#836 — index entries differing only in case are now
  genuinely folded by default: merged under the lowercase headword with the
  union of their subentries and locators, so nothing is lost. (Also fixes
  the same silent loss between entries differing only in markup, and in
  see/see-also lists.)
- metanorma/metanorma-standoc#1234 — new document attribute
  `:index-case-sensitive:`, which suppresses the folding and keeps
  case-distinct headwords apart, sorted adjacently, as in the SMRL-style
  index shown above as the expected output.
- metanorma/metanorma-bipm#661 — the same rendering fix mirrored in the
  BIPM per-letter index (gated on the next isodoc release).

For this document class, stepmod2mn should set `:index-case-sensitive:` in
the documents it generates; that restores the legacy SMRL behaviour, with
`Activity` (ARM object definition, mapping specification, ARM EXPRESS-G)
and `activity` (term) as separate headwords. Without the attribute, the
default now yields a single merged `activity` entry carrying all four
subentries — no more dropped locators either way.

The attribute will be documented on metanorma.org (author attribute
reference) as of metanorma-standoc v3.5.0.

cc @manuelfuenmayor

🤖
